crates/sui-rpc-benchmark/src/direct/query_executor.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// This module executes enriched benchmark queries against the database.
+/// Each query's execution is timed and recorded via MetricsCollector.
+/// And the results are aggregated and reported via BenchmarkResult.
+use std::time::Instant;
+
+use anyhow::Result;
+use bb8::Pool;
+use bb8_postgres::PostgresConnectionManager;
+use rand::seq::SliceRandom;
+use rand::SeedableRng;
+use sui_indexer_alt_framework::task::TrySpawnStreamExt;
+use tokio_postgres::{types::ToSql, NoTls};
+use tracing::info;
+use url::Url;
+
+use crate::direct::benchmark_config::BenchmarkConfig;
+use crate::direct::metrics::{BenchmarkResult, MetricsCollector};
+use crate::direct::query_enricher::{EnrichedBenchmarkQuery, SqlValue};
+
+pub struct QueryExecutor {
+    pool: Pool<PostgresConnectionManager<NoTls>>,
+    enriched_queries: Vec<EnrichedBenchmarkQuery>,
+    config: BenchmarkConfig,
+    metrics: MetricsCollector,
+}
+
+impl QueryExecutor {
+    pub async fn new(
+        db_url: &Url,
+        enriched_queries: Vec<EnrichedBenchmarkQuery>,
+        config: BenchmarkConfig,
+    ) -> Result<Self> {
+        let manager = PostgresConnectionManager::new_from_stringlike(db_url.as_str(), NoTls)?;
+        let pool = Pool::builder().build(manager).await?;
+
+        Ok(Self {
+            pool,
+            enriched_queries,
+            config,
+            metrics: MetricsCollector::default(),
+        })
+    }
+
+    async fn worker_task(
+        pool: Pool<PostgresConnectionManager<NoTls>>,
+        enriched_queries: Vec<EnrichedBenchmarkQuery>,
+        metrics: MetricsCollector,
+        deadline: Instant,
+    ) -> Result<()> {
+        let client = pool.get().await?;
+        let mut query_rng = rand::rngs::StdRng::from_entropy();
+        let mut row_rng = rand::rngs::StdRng::from_entropy();
+        while Instant::now() < deadline {
+            let enriched = enriched_queries
+                .choose(&mut query_rng)
+                .ok_or_else(|| anyhow::anyhow!("No queries available"))?;
```

Comment:
```suggestion
                .context("No queries available")?;
```

Reply:
it's an option so no original `Result` to add additional context to?

Reply:
`Context` works on `Option` too (to convert it into an `anyhow::Result`).

Reply:
yes, I saw your PR and will amend this change to the latest commit, TIL, thanks!